libs/server/Objects/Hash/HashObjectImpl.cs

```diff
@@ -257,7 +260,7 @@ private void HashRandomField(byte* input, int length, ref SpanByteAndMemory outp
                     countDone = count;
 
                     // Prepare response
-                    if (!Int32.TryParse(Encoding.ASCII.GetString(countParameterByteArray), out var countParameter))
+                    if (!Utf8Parser.TryParse(countParameterByteArray, out int countParameter, out _, default))
```

Comment:
I like the use of `Utf8Parser` for this. However, we should check the `bytesConsumed` as well if we are consuming any unsanitized user input. `Utf8Parser.TryParse` will match substrings - so this could have unintended side-effects for invalid strings such as `123abs`.

This applies to all the `Utf8Parser`-related changes in this PR.

Reply:
Ah very good catch! Will restore original behavior by checking we consumed the entire buffer which is fed into U8.TryParse :+1: